### PR TITLE
api: refactor config API to adapt tidb-operator/TOML changes

### PR DESCRIPTION
**Pull request by xhebox**
Signed-off-by: xhe <xw897002528@gmail.com>

### What problem does this PR solve?

Issue Number: ref #161 

Problem Summary: Due to we have adapt `ConfigWrapper` in `tidb-operator`, it becomes harder to set config for each component separately. Further more, we need to issue two requests for each instace. Instead, we just upload the whole config, and pick updatable part now.

What is changed and how it works:

1.  `config [log | proxy] [get | put]` is replaced by one single command `tiproxyctl config get/put`
2. Watch channels for `XXXOnlineConfig` are replaced by `chan *config.Config`
3. Init values are inited in `ConfigManager.Init()`
4. Stablize log config by removing file size check

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Notable changes

- [ ] Has configuration change
- [x] Has HTTP API interfaces change (Don't forget to [add the declarative for API](https://github.com/tikv/pd/blob/master/docs/development.md#updating-api-documentation))
- [x] Has tiproxyctl change
- [ ] Other user behavior changes

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: pkg/manager/logger/manager_test.go [hunk 1]
```diff
@@ -78,10 +80,7 @@ func TestUpdateCfg(t *testing.T) {
 				}
 			},
 			check: func(files []os.FileInfo) bool {
-				if len(files) != 1 {
-					return false
-				}
-				return files[0].Size() >= int64(2500*1024)
```
Comment (xhebox) [inline pkg/manager/logger/manager_test.go:84]: Rotation logger is not going to flush all data immediately into files, making file size always a little bit less than the expected value. Just remove it.

Comment (djshow832) [reply pkg/manager/logger/manager_test.go:84]: But I saw you have added `log.Sync()` to flush it.
Besides, each case waits for at most 3 seconds.

Comment (xhebox) [reply pkg/manager/logger/manager_test.go:84]: Yeah, there is `log.Sync()`. But it is stub, `lumberjack` does not have a `sync()`. Check https://github.com/pingcap/TiProxy/blob/main/lib/util/cmd/syncer.go#L42-L48 .

Considering it is always failing and `lumberjack` is actually a well-tested library. I think it is just fine not to test the size behavior.

File: pkg/manager/logger/manager.go [hunk 2]
```diff
@@ -51,20 +51,23 @@ func NewLoggerManager(cfg *config.Log) (*LoggerManager, *zap.Logger, error) {
...
 
-func (lm *LoggerManager) watchCfg(ctx context.Context, cfgCh <-chan *config.LogOnline) {
+func (lm *LoggerManager) watchCfg(ctx context.Context, cfgch <-chan *config.Config) {
 	for {
 		select {
 		case <-ctx.Done():
 			return
-		case cfg := <-cfgCh:
+		case acfg := <-cfgch:
+			cfg := &acfg.Log.LogOnline
```
[the remark this answers is not in the document]
Comment (xhebox) [reply pkg/manager/logger/manager.go:69]: No delete in the current implementation. And it is ensured to be valid pointer:

```
	var c config.Config
		if k.Type == KVEventDel {
			ch <- &c
		} else {
```

File: pkg/manager/logger/manager_test.go (hunk 1 above)
Comment (djshow832) [reply pkg/manager/logger/manager_test.go:84]: It never failed before. Does it fail in this PR?

Comment (xhebox) [reply pkg/manager/logger/manager_test.go:84]: Yes, very stable failure on my machine. I guess it is still not stable.

Comment (djshow832) [reply pkg/manager/logger/manager_test.go:84]: Does it fail on your machine before this PR?

Comment (xhebox) [reply pkg/manager/logger/manager_test.go:84]: No, but I can't figure out any difference. I've debugged that `config options` are updated correctly. It is just failing with the correct test case.

Comment (djshow832) [reply pkg/manager/logger/manager_test.go:84]: I'll try it on my machine. I'm afraid there's some other bug.

Comment (xhebox) [reply pkg/manager/logger/manager_test.go:84]: BTW, it failed on CI before.

**Review by djshow832 — APPROVED** [review]

**Pull request merged**